ui/src/ui-utils.ts

```diff
@@ -19,3 +19,45 @@ export function isKeypressEscape(event: React.KeyboardEvent) {
   }
   return false;
 }
+
+const PATTERN_NON_WORD_CHAR = /\W/;
+const PATTERN_WORD_CHAR = /\w/;
+const ELLIPSIS = '…';
+
+/**
+ * Truncates the provided text such that no more than limit characters are rendered and adds an
+ * ellipsis upon truncation by default.  If the text is shorter than the provided limit, the full
+ * text is returned.
```

Comment:
Could we throw in a link back to the S2 code this is based on? Just to make it easier for me to find the implementation and keep it updated while we're waiting for standardized S2 components.

Reply:
Yes, good call!

Reply:
I added a sufficiently opaque indication of where this was derived from!